ts/src/huobi.ts

```diff
@@ -2931,27 +3026,40 @@ export default class huobi extends Exchange {
 
     networkIdToCode (networkId, currencyCode = undefined) {
         // here network-id is provided as a pair of currency & chain (i.e. trc20usdt)
-        const keys = Object.keys (this.options['networkNamesByChainIds']);
-        const keysLength = keys.length;
-        if (keysLength === 0) {
-            throw new ExchangeError (this.id + ' networkIdToCode() - markets need to be loaded at first');
+        if (!('networkNamesByChainIds' in this.options)) {
+            if (!this.currencies) {
+                // currencies need to be loaded, otherwise it's impossible to work
+                throw new ExchangeError (this.id + ' networkIdToCode() - markets need to be loaded at first');
+            }
+            this.options['networkNamesByChainIds'] = {};
```

Comment:
isn't a bit odd that we're populating this structure inside this method?

Reply:
it's a bit odd yes, but as said, I see this as a temporary workaround in huobi, before other network-related PRs will be merged.
otherwise, I have to poison `fetchCurrencies` method with such things, and I think it's better to keep `fetchCurrencies` clean and just embed this code in this overriden `network-id/code` overloads (which will be removed at all when other network-related PRs will be merged). 
also, this code will only be executed once per instance lifetime.

Reply:
@ttodua I think we might be overcomplicating the networks here, as you know huobi returns currency+chain name together right?

If we want to parse from networkCode to network id, can't we just append together the currencyId + networkId (usdt+erc20) and do the inverse to translate the networkId to network code? We can even populate networksById programmatically inside fetchCurrencies with 
```
{
  'erc20usdd': 'ERC20',
  'solusdt' : SOL, 
  'etc'
}
```

and networks with 

```
{
    "ERC20": 'erc20',
    "TRC20: 'trc20',
}
```

then `networkIdToCode` could make an access direct to the `networkIds` mapping, and `networkCodeToId` could get the networkName from `networks` and append the currencyId

Reply:
@ttodua you're right regarding the logic, it's not consistent so we can't do that

Reply:
take `nft, soc, usdj, rock, pearl, sun...` currencies and try to withdraw on TRON network.

Reply:
and that above comment only refers to `TRON` while for `eth` network there are tens of times more examples of random ids.

Reply:
@ttodua I agree that the names are a bit random and can't be guessed, but two things are still true, first
networkId -> networkCode is still a 1 to 1 relationship so `networksById` could be populated normally and programmatically,

second,  networkCode to networkId is still possible using the networkCode + currencyCode, we just need to store it differently (maybe add another level to `networks` for instance`

Reply:
@ttodua 

1) just saying that given a network-id (trc20usdt or usdterc20) you can map to a networkCode 

2) Regarding the `pear/sun/etc`, those are exceptions to the rule that need to be handled one way or the other (not sure what huobi requires in these cases), but we should not overcomplicate one design just to encapsulate these weird choices from some exchanges when the simple version fits 90% of the cases. 

![image](https://github.com/ccxt/ccxt/assets/43336371/c2814e78-ec80-4835-bae3-725080152b18)

Reply:
1) you are still referring to `usdt`, please put it aside it for a while and look to any other currency. my listed other coins are not exceptional, they are common. i've just listed small amount of TRON, but will list much more for eth, rpobably 10 times more

Reply:
huobi is not exclusion, there are some other exchanges that have something like that.

Reply:
but hey  - sure ✌️ i am here to follow whatever you decide. just confirm me the next step and I'll recode huobi with replacements/basechain approach if you think so

Reply:
but before you'd decide to still go with the route you should also look into not only TRON, but any other network based currencies,, which have more than 2 network entries (like `usdd, gmx, id` or any other).

Reply:
@ttodua Don't worry I'm not saying that what you did is wrong, for now I'm just poking around and making sure we're following the simplest possible approach, but I haven't had yet time to review and test it properly